### ISSUE TITLE
Parse shapes.txt in worker thread

### ISSUE THREAD

**Issue by jayenashar** [thatsmydoing/gtfs.html#21]
Hi there,

Is it possible you could parse shapes.txt in a worker thread, so that the website can be used while it is being parsed?  I have some very large GTFS files, and shapes.txt is the largest portion, and it takes a few minutes for papaparse to parse.

Thanks,
Jayen

**Comment by thatsmydoing** [comment]
Yeah, it doesn't really deal too well with large GTFS files. I'll try your suggestion of moving it off to a worker thread. I'm also thinking of delaying parsing of the shapes since that's relatively isolated in terms of how it's related to everything else.

**Comment by jayenashar** [comment]
i was looking at papaparse as well, and i see if disables fastmode if the csv has quotes.  i've got my own fast csv parser in java (which handles quotes), but i'm not sure how to make it work with javascript streams.  i could look more at that, if offloading shapes to a worker thread doesn't work.

**Comment by thatsmydoing** [comment]
I've pushed a fix that's defers shape parsing. There's still a delay once it pushes data through the worker / main thread barrier though.

On a different note, is it possible for you to send me the gtfs file you're using?

If you self-host, you can actually make the server do the parsing. This is currently also in nodejs though, so I'm not sure how much faster that will be.

**Comment by jayenashar** [comment]
https://opendata.transport.nsw.gov.au/sites/default/files/resources/full_greater_sydney_gtfs_static_0.zip

yours seems to be the only free one that works browser side and doesn't require me to be a transport operator registered with google, so that's why i'm using your software.  if i were self-hosting, i'd probably use someone else's.  (no offense intended)

**Comment by jayenashar** [comment]
sorry i forgot to mention i'm using https://gtfs.pleasantprogrammer.com/ .  is that where you pushed to?

you may consider updating your github project description to point to the URL so people know they don't have to host themselves.

**Comment by jayenashar** [comment]
thanks.  i can see this and the other changes.
